Improvements for Intent.CreateChooser

andrk0p opened the pull request:
### Description
I found that the selection of photos or videos does not work on all devices.

### Actual Behavior

![photo_2021-09-03_19-03-29 (2)](https://user-images.githubusercontent.com/26412172/132035257-d2961410-f0b3-4d6f-a171-8850d388a935.jpg)


### Expected behavior

![photo_2021-09-03_19-03-29](https://user-images.githubusercontent.com/26412172/132035363-7acce9a4-7401-462f-9cec-cd01a2ec3bc6.jpg)

The chooserIntent object in all testing cases returns Android.App.Result.Canceled in the onActivityResult method.

I couldn't find a way to get chooserIntent to work correctly on my device, so I added a check for the operating system version to rule out this problem on similar devices.

- Fixed the PickAsync method for devices with Android 11

### API Changes

Added:

```
public bool UseCreateChooser { get; set; } = true;
```

### Configuration
 - Device: Xiaomi Redmi Note 8 Pro
 - OS Version: MIUI Global 12.5.1 Android 11 RP1A
 - Plugin Version: 1.0.2.1

### PR Checklist ###

- [x] All projects build
- [x] Has samples
- [x] Rebased onto current main

dimonovdd commented on MediaGallery/MediaGallery/PickMedia.android.cs:
```diff
@@ -62,7 +61,9 @@ static async Task<IEnumerable<IMediaFile>> PlatformPickAsync(MediaPickRequest re
                     intent.PutExtra(Intent.ExtraTitle, request.Title);
 
-                chooserIntent = Intent.CreateChooser(intent, request.Title ?? string.Empty);
+                intent = !Platform.HasSdkVersion(30)
```
Thanks. Сan we add a check based on the device model? I tested this plugin on Android 11 devices and it works correctly.

andrk0p replied:
Yes, I can add a device model check using ```Xamarin.Essentials.DeviceInfo.Model```

dimonovdd commented:
@andrk0p I was able to test this on a device like yours but with 12.5.2 version of MIUI. I can't reproduce this problem. Do you have any ideas?

andrk0p commented:
> 
> 
> @andrk0p I was able to test this on a device like yours but with 12.5.2 version of MIUI. I can't reproduce this problem. Do you have any ideas?

I don't have any ideas. Maybe the problem is in MAUI 12.5.1

dimonovdd commented:
@andrk0p Maybe, can we add a new parameter for determining use of `Intent.CreateChooser`?

andrk0p commented:
> 
> 
> @andrk0p Maybe, can we add a new parameter for determining use of `Intent.CreateChooser`?

I added the NeedUseCreateChooser parameter and updated the usage example

dimonovdd commented on README.md:
```diff
@@ -130,12 +131,14 @@ protected override void OnActivityResult(int requestCode, Result resultCode, Int
 - When using the `PickAsync` method the `selectionLimit` parameter just sets multiple pick allowed
 - A request to cancel `PickAsync` method will cancel a task, but the picker UI can remain open until it is closed by the user
 - The use of `Title` property depends on each device
+- `UseCreateChooser` specifies whether to use Intent.createChooser for picking.
```
```suggestion
- `UseCreateChooser` specifies whether to use `Intent.CreateChooser`
```

dimonovdd commented on MediaGallery/MediaGallery/PickMedia.android.cs:
```diff
@@ -62,7 +62,11 @@ static async Task<IEnumerable<IMediaFile>> PlatformPickAsync(MediaPickRequest re
+                if (request.UseCreateChooser)
+                {
+                    intentDisposable = intent;
+                    intent = Intent.CreateChooser(intentDisposable, request.Title ?? string.Empty);
```
```suggestion
                    intent = Intent.CreateChooser(intent, request.Title ?? string.Empty);
```

dimonovdd commented on MediaGallery/MediaGallery/PickMedia.android.cs:
```diff
@@ -26,7 +26,7 @@ static async Task<IEnumerable<IMediaFile>> PlatformPickAsync(MediaPickRequest re
             token.ThrowIfCancellationRequested();
             Intent intent = null;
-            Intent chooserIntent = null;
+            Intent intentDisposable = null;
```
```suggestion
            IDisposable intentDisposable = null;
```

dimonovdd commented on MediaGallery/MediaPickRequest/MediaPickRequest.shared.cs:
```diff
@@ -28,5 +28,8 @@ public MediaPickRequest(int selectionLimit = 1, params MediaFileType[] types)
         /// <summary>Gets or sets the source rectangle to display the Share UI from. This is only used on iPad currently.</summary>
         public Rectangle PresentationSourceBounds { get; set; } = default;
+
+        /// <summary>Gets or sets whether to use Intent.CreateChooser for picking. Currently used only for Android.</summary>
```
```suggestion
        /// <summary>Gets or sets whether to use Intent.CreateChooser. Currently used only for Android.</summary>
```

dimonovdd reviewed (changes requested):
LGTM! I added a few minor edits:
